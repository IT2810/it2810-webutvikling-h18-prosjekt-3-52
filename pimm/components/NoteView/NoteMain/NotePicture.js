import React from 'react';
import { Image, Button, View } from 'react-native';
import { ImagePicker, Permissions } from 'expo';

export default class NotePicture extends React.Component{
    state = {
        image: null,
        permission: false,
    };

    render() {
        let { image } = this.state;


        if (image === null) {
            return (
                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                    <Button
                        title="Pick an image that fits to your note"
                        onPress={this.pickImage}
                        color="#B8860B"
                    />

                    <View style={{ borderColor: 'black', borderWidth: 1, backgroundColor: 'beige', width: 280, height: 300 }}>
                        
                    </View>
                </View>
            );
        } else {
            return (
                <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
                    <Button
                        title="Pick an image that fits to your note"
                        onPress={this.pickImage}
                        color="#B8860B"
                    />

                    <View style={{ borderColor: 'black', borderWidth: 1, backgroundColor: 'beige', width: 280, height: 300 }}>
                        <Image source={{ uri: image }} style={{ width: 280, height: 300 }} />
                    </View>
                </View>
            );
        }


    }

    componentWillReceiveProps(nextProps){
        if (this.props.image!==nextProps.image || nextProps.fl) {
            this.setState({image: nextProps.image});
            this.props.sfl();
        }
    }


    pickImage = async () => {
        if (this.state.permission === false) {
            const { status } = await Permissions.askAsync(Permissions.CAMERA_ROLL);
            if(status === "granted"){
                this.setState({permission: true});
            }

        }
        if (this.state.permission === true) {
            let result = await ImagePicker.launchImageLibraryAsync({
                allowsEditing: false,
            });

            if (!result.cancelled) {
                this.setState({ image: result.uri });
                this.props.sendImageUp(result.uri);

            }
        }
    };
}