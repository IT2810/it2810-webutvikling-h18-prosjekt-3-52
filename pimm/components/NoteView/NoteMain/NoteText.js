import React from 'react';
import { Text, TextInput, View } from 'react-native';

export default class NoteText extends React.Component{
	constructor(props){
		super(props);
		this.state={
			note: "",
		};
		
	}
	
	changeNote(newNote){
		this.setState({note: newNote});
	
	}
	
	handleChange(newNote){
		this.changeNote(newNote);
		this.props.sendNoteUp(newNote);
	}
	
	componentWillReceiveProps(nextProps){
		if(this.props.note!==nextProps.note){
			this.setState({note: nextProps.note});
		}
	}
	
	render() {
		return(
			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center', padding: 5 }}>
			<TextInput 
			placeholder="Write note here..."
			multiline = {true}
			style={{padding: 5, width: 280, backgroundColor: 'beige'}}
			onChangeText={(note) => this.handleChange(note)}
			value={this.state.note}
			/>
			</View>		
		);
	}
	
}