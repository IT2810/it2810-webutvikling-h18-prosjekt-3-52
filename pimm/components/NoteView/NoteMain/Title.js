import React from 'react';
import { Text, TextInput, View, Button, AsyncStorage } from 'react-native';
import SaveButton from '../NoteTopBar/SaveButton.js';

export default class Title extends React.Component{
	constructor(props){
		super(props);
		this.state={
			title: "",
		};
		
	}
	
	changeTitle(newTitle){
		this.setState({title: newTitle});
	
	}
	
	handleChange(newTitle){
		this.changeTitle(newTitle);
		this.props.sendTitleUp(newTitle);
	}
	
	componentWillReceiveProps(nextProps){
		if(this.props.title!==nextProps.title || nextProps.fl){
			this.setState({title: nextProps.title});
			this.props.sfl();
			
		}
	}
	
	
	
	render() {
		return(
			<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center', padding: 5 }}>
			<TextInput 
			placeholder="Write title here"
			style={{padding: 10, width: 280, backgroundColor: 'beige', fontWeight: 'bold', fontSize: 20}}
			onChangeText={(title) => this.handleChange(title)}
			value={this.state.title}
			/>
			</View>		
		);
	}
	
}