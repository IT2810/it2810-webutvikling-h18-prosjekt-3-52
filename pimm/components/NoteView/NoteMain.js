import React from 'react';
import Title from './NoteMain/Title.js';
import NoteText from './NoteMain/NoteText.js';
import NotePicture from './NoteMain/NotePicture.js';
import { View, Button } from 'react-native';

export default class NoteMain extends React.Component{
	constructor(props){
		super(props);
		this.state={
			title: "",
			note: "",
			image: "",
		}
	}
	
	
	render(){
		return(
		<View>
		<Title sendTitleUp={this.props.sendTitleUp} title={this.props.title}/>
		<NoteText sendNoteUp={this.props.sendNoteUp} note={this.props.note}/>
		<NotePicture sendImageUp={this.props.sendImageUp} image={this.props.image}/>
		</View>
		
		);
	}
}