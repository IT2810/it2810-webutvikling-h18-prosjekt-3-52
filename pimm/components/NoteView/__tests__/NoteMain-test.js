import React from 'react';
import 'react-native';
import NoteMain from '../NoteMain.js';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';


test('NoteMain renders correctly', () => {
  const nm = renderer.create(<NoteMain />).toJSON();
  expect(nm).toMatchSnapshot();
  expect(nm.type).toBe("View");
  expect(nm.children[0].type).toBe("View");
  expect(nm.children[1].type).toBe("View");
  expect(nm.children[2].type).toBe("View");
  
});

/*test('Props are sent', ()=>{ TODO: Flytt denne testen til App.
	const renderer = new ShallowRenderer();
	renderer.render(<NoteMain />);
	const result = renderer.getRenderOutput();
	expect(result.props.children[0].props.sendTitleUp).toBeDefined();
	expect(result.props.children[1].props.sendNoteUp).toBeDefined();
	expect(result.props.children[2].props.sendImageUp).toBeDefined();
	

});*/
