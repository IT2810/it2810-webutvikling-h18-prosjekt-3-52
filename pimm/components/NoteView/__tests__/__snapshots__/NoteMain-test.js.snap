// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteMain renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Write title here"
      style={
        Object {
          "backgroundColor": "beige",
          "fontSize": 20,
          "fontWeight": "bold",
          "padding": 10,
          "width": 280,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "padding": 5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      multiline={true}
      onChangeText={[Function]}
      placeholder="Write note here..."
      style={
        Object {
          "backgroundColor": "beige",
          "padding": 5,
          "width": 280,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityLabel={undefined}
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          disabled={undefined}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#B8860B",
              },
            ]
          }
        >
          Pick an image that fits to your note
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "beige",
          "borderColor": "black",
          "borderWidth": 1,
          "height": 300,
          "width": 280,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": null,
          }
        }
        style={
          Object {
            "height": 300,
            "width": 280,
          }
        }
      />
    </View>
  </View>
</View>
`;
